services/parser-database/parser.py

```diff
@@ -0,0 +1,218 @@
+        if (pdf_text[i] == "artículo" or pdf_text[i] == "articulo") and (
+                pdf_text[i + 1] == str(article_count) + ".-"
+                or pdf_text[i + 1] == str(article_count) + "-"
+                or pdf_text[i + 1] == str(article_count) + "."):
+            logging.info("Article #" + str(article_count) + " recognized!")
+            articles.append(Article(article_count, article_text))
+            article_text = ""
+            article_count += 1
+            i += 1
+        else:
+            article_text += " " + pdf_text[i]
+        i += 1
+    return articles
+
+
+mty_document = {
+    "hash":
+    "afafbfbdce8c40924edae00f6ce54f0c639ce42a2" +
+    "c0fbbfa6ab82ea6925827c51",
+    "jurisdiction":
+    "Monterrey",
+    "url":
+    "http://www.guadalupe.gob.mx/wp-content/up" +
+    "loads/2019/09/Nuevo-Reglamento-Homologado-1.pdf",
+}
+document_list = []
+document_list.append(mty_document)
+
+
+def parse_all_documents():
+    """Parses all documents that are specified on the DB
+    (For obvious reasons, this won't work as is while the DB is not in use)
+    (But will still parse the hardcoded document)"""
+    for document in document_list:
+        parse(document)
+
+
+def has_file_changed(past_hash):
+    """Sees if the file is different.
+
+    Args:
+        past_hash (string): hash to compare from
+
+    Returns:
+        [boolean]: [if the file has changed or not]
+    """
+    hasher = hashlib.sha256()
+    with open("regs.pdf", "rb") as pdf_file:
+        file_buffer = pdf_file.read()
+        hasher.update(file_buffer)
+        if hasher.hexdigest() == past_hash:
+            return False
+        return True
+
+
+# TODO Change to point to Javier's service
+def get_keywords(text):
+    """Get keywords that relate to this article (Javier's service)
+
+    Args:
+        text (sting): text to extract keywords from
+
+    Returns:
+        [JSON]: extracted keywords
+    """
+    """
+    return requests.post(
+        "localhost:8000", params={"text": text}
+    ).json()
+    """
+    keywords_service_response = keywordmock.get_keywords(text)
+    return json.loads(keywords_service_response)
+
+
+def article_to_dictionary(article):
+    """Converts an parsed article to a dictionary and saves it
+
+    Args:
+        article (Article): [article object]
+    """
+    article_dict = {
+        "articleNumber": article.number,
+        "text": article.text,
+        "wordCount": len(article.text.split())
+    }
+    articles_in_memory.append(article_dict)
+    save_keywords_in_memory(get_keywords(article.text), article)
+
+
+def save_keywords_in_memory(keywords, article):
+    """Saves the keywords from an article in memory
+
+    Args:
+        keywords (JSON): contains keywords
+        article (Article): article object
+    """
+    split_article = article.text.split()
+    for keyword in keywords["keywords"]:
+        frequency = split_article.count(keyword)
+        if keyword not in keywords:
```

Comment:
I assume you meant `if keyword not in split_article` since you're looping through all the keywords in `keywords`?

Either way, I would recommend not adding blank entries in your dictionary, cause that could take up unnecessary memory for words you may never use again.

Reply:
no, if the keyword that is being added does not exist in the list of keywords, it will create an empty list that will be populated in line 195, if the keyword exists, it will not create an empty list. 

The `split_article` allows the program to search for the frequency of the word in the text.